packages/core-p2p/lib/server/versions/1/handlers.js

```diff
@@ -297,13 +297,23 @@ exports.postTransactions = {
    * @return {Hapi.Response}
    */
   async handler (request, h) {
+    const blockchain = container.resolvePlugin('blockchain')
+
+    if (!blockchain) {
+      return h.response({ success: false, error: 'Blockchain not ready' }).code(500)
```

Comment:
This is already handled by a plugin. https://github.com/ArkEcosystem/core/blob/develop/packages/core-p2p/lib/server/plugins/blockchain-ready.js